Codes/Brawlback/Brawlback.h

```diff
@@ -57,114 +51,81 @@ inline void updateGamePadSystem() { updateGame(PAD_SYSTEM); }
 
 u32 getCurrentFrame();
 
-
-enum PlayerType : u8
+struct PlayerSettingsImpl
 {
-    PLAYERTYPE_LOCAL = 0x0,
-    PLAYERTYPE_REMOTE = 0x1,
+    PlayerSettings _playerSettings;
 };
 
-struct PlayerSettings
+struct GameSettingsImpl
 {
-    u8 charID;
-    u8 charColor;
-    PlayerType playerType;
-    u8 controllerPort;
-    u16 nametag[NAMETAG_SIZE];
-    u8 displayName[DISPLAY_NAME_SIZE];
-    u8 connectCode[CONNECT_CODE_SIZE];
+    GameSettings _gameSettings;
 };
 
-struct GameSettings
+struct PreserveBlockImpl
 {
-    u8 localPlayerIdx;
-    u8 numPlayers;
-    u16 stageID;
-    u32 randomSeed;
-    PlayerSettings playerSettings[MAX_NUM_PLAYERS];
-};
-
-struct PreserveBlock {
-    u32 address;
-    u32 length;
+    PreserveBlock _preserveBlock;
 };
 
-#pragma pack(push, 2)
-
-struct BrawlbackPad {
-    unsigned short buttons;
-    char stickX;
-    char stickY;
-    char cStickX;
-    char cStickY;
-    char LTrigger;
-    char RTrigger;
-
-    BrawlbackPad() {
-        buttons = 0;
-        stickX = 0;
-        stickY = 0;
-        cStickX = 0;
-        cStickY = 0;
-        LTrigger = 0;
-        RTrigger = 0;
+struct BrawlbackPadImpl {
+    BrawlbackPad _brawlbackPad;
+
+    BrawlbackPadImpl()
+    {
+        _brawlbackPad.buttons = 0;
+        _brawlbackPad.stickX = 0;
+        _brawlbackPad.stickY = 0;
+        _brawlbackPad.cStickX = 0;
+        _brawlbackPad.cStickY = 0;
+        _brawlbackPad.LTrigger = 0;
+        _brawlbackPad.RTrigger = 0;
     }
 };
-#pragma pack(pop)
 
-#pragma pack(push, 4)
+struct PlayerFrameDataImpl {
```

Comment:
What is the point of these Impl structs?

Reply:
I see that some have constructors implemented.. I assume that's why? Might as well just have a `PlayerFrameData defaultPlayerFramedata();` function or something like that to return a strcuture initialized how you want. I think it adds complexity to have all of these wrapper classes.